.github/workflows/macos_tests.yml

```diff
@@ -0,0 +1,41 @@
+name: macos_tests
+on: [pull_request, push]
+jobs:
+  macos_tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - 2.7.18
+          - 3.5.10
+          - 3.6.15
+          - 3.7.10
+          - 3.8.10
+          - 3.9.5
+          - 3.10.0
+    runs-on: macos-10.15
+    steps:
+      - uses: actions/checkout@v2
+      # Normally, we would use the superbly maintained...
+      # - uses: actions/setup-python@v2
+      #   with:
+      #    python-version: ${{ matrix.python-version }}
+      # ... but in the repo, we want to test pyenv builds on macOS
+      - run: |
+          brew install autoconf openssl@1.1 pkg-config readline
```

Comment:
Is autoconf needed? Python releases include already-generated configure files.
What about other dependencies as per https://github.com/pyenv/pyenv/wiki#suggested-build-environment?

Newer release lines should be tested in newer MacOS and with OpenSSL 3 but this is good enough for a start I guess.

Reply:
Those are what Homebrew listed as dependencies, so I just copied them in here. 

https://github.com/Homebrew/homebrew-core/blob/b3716bffc17f953c487472d0e5a0e9ae230e7094/Formula/pyenv.rb#L24-L33

If you prefer what’s on the wiki I can easily change it.

Reply:
We need to test the suggested build environment so please do.

Reply:
All set

Reply:
Oh, there something I noticed but forgot to mention. It looks like even though the Homebrew versions of sqlite3, xz, and zlib are installed, pyenv is using the native macOS versions. It makes me wonder if the wiki should be more specific about what versions of macOS require those dependencies.

Reply:
Sure, it would help if someone looked into this. I know that previous MacOS versions didn't have Zlib -- this is probably the case with other libs, too. Maybe for non-EOL MacOS versions,  some of these are no longer needed.